apollo-router/src/plugins/telemetry/mod.rs

```diff
@@ -985,6 +1005,65 @@ impl Telemetry {
         )
     }
 
+    fn get_cache_attributes(
+        subgraph_name: Arc<String>,
+        sub_request: &Request,
+    ) -> Option<CacheAttributes> {
+        let body = sub_request.subgraph_request.body();
+        let hashed_query = hash_request(body);
+        let representations = body
+            .variables
+            .get(REPRESENTATIONS)
+            .and_then(|value| value.as_array())?;
+
+        let keys = extract_cache_attributes(representations).ok()?;
+
+        Some(CacheAttributes {
+            subgraph_name,
+            headers: sub_request.subgraph_request.headers().clone(),
+            hashed_query,
+            representations: keys,
+        })
+    }
+
+    fn update_cache_metrics(sub_response: &SubgraphResponse, cache_attributes: CacheAttributes) {
+        let mut vary_headers = sub_response
+            .response
+            .headers()
+            .get_all(header::VARY)
+            .into_iter()
+            .filter_map(|val| {
+                val.to_str().ok().map(|v| {
+                    v.to_string()
+                        .split(", ")
+                        .map(|s| s.to_string())
+                        .collect::<Vec<String>>()
+                })
+            })
+            .flatten()
+            .collect::<Vec<String>>();
+        vary_headers.sort();
+        let vary_headers = vary_headers.join(", ");
+
+        let hashed_headers = if vary_headers.is_empty() {
+            String::new()
+        } else {
+            hash_vary_headers(&cache_attributes.headers)
+        };
+
+        for (typename, hashed_representation) in cache_attributes.representations {
+            ::tracing::info!(
```

Comment:
I don't think this is a good approach. The cardinality will be too high and we risk leaking memory as prom keeps metrics around forever.
Suggest using counting bloom filters to simulate checking for a cache hit. The count from the filter can be fed into a histogram metric without the hashed representations, we'll be able to see the distribution of what could be cached and what couldn't be.

Reply:
fixed in https://github.com/apollographql/router/pull/3532/commits/52a306f1c915670fd2c0248ff0af3807018d516f